R/qenv-get_code.R

```diff
@@ -23,13 +24,19 @@ setGeneric("get_code", function(object, deparse = TRUE) {
 
 #' @rdname get_code
 #' @export
-setMethod("get_code", signature = "qenv", function(object, deparse = TRUE) {
+setMethod("get_code", signature = "qenv", function(object, deparse = TRUE, names = NULL) {
   checkmate::assert_flag(deparse)
-  if (deparse) {
-    format_expression(object@code)
+  code <- if (!is.null(names)) {
+    get_code_dependency(object, names)
   } else {
     object@code
   }
+  if (code[1] == "") code <- code[-1]
```

Comment:
```
> get_code(q, deparse = FALSE, names = "x") 
Error in if (code[1] == "") code <- code[-1] : 
  missing value where TRUE/FALSE needed
In addition: Warning message:
In get_code_dependency(object, names) :
  Objects not found in 'qenv' environment: x
```

`code` is `character(0)` so `==` returns `NA` and that crashes `if`

Reply:
nice, thanks. appreciate! duly noted

Reply:
https://github.com/insightsengineering/teal.code/pull/146/commits/b619c3b9eca10d5fae0c8499cbe635ac44d97c9d